### DISCUSSION TITLE
u-blox Receiver Type Query and Supported Bands Output

### DISCUSSION THREAD

**Discussion by MalikD233** [semuconsulting/pyubx2#143] [Ideas and Feature Requests]
Dear Semuconsulting Team,

I would like to suggest the addition, if not already present, of a feature that enables users to query the u-blox receiver type. Additionally, it would be beneficial to incorporate a function that provides an output detailing the supported bands (L1, L2, or L5) of the utilized u-blox receiver.

Thank you in advance for considering this enhancement. Your dedication to excellence is truly commendable.

Warm regards.

**Comment by semuadmin** [comment]
Hi @MalikD233 ,

This is all existing functionality - it simply requires knowledge of the relevant UBX messages to poll, which you can get from the [Integration and Interface Manuals](https://www.u-blox.com/en/positioning-chips-and-modules) for your particular u-blox receiver.

Re. "receiver type" - This information can be obtained by polling the UBX MON-VER, MON-HW and MON-RF message types - this is exactly what the [PyGPSClient](https://github.com/semuconsulting/PyGPSClient) utility does to produce the following receiver information panel:

<img width="694" alt="receiver info" src="https://github.com/semuconsulting/pyubx2/assets/28569967/1cea445e-d316-4de9-9c49-4e6dbca5fa09">

See [this code example](https://github.com/semuconsulting/pyubx2/blob/master/examples/ubxpoller.py) for an illustration of how to poll specific UBX message types.

Re. "supported bands" - This information can be obtained by polling the UBX MON-RF message type, though the specific configuration (e.g. L2/L5) is defined in hardware rather than software - see your receiver's Integration Manual for details. Some advanced receivers (e.g. ZED-F9P) also support the UBX MON-SPAN message which will give you real time spectrum analysis data for each supported radio frequency band - in this example from a ZED-F9P module, there is support for L1 and L2 bands:

<img width="640" alt="mon-span spectrum" src="https://github.com/semuconsulting/pyubx2/assets/28569967/ffdd5a82-0255-4933-b8d9-339f0591151c">

See [this code example](https://github.com/semuconsulting/pyubx2/blob/master/examples/mon_span_spectrum.py) for an illustration of how to produce a spectrum plot like the one above from a raw MON-SPAN message log.
